tests/utils/test_utils.py

```diff
@@ -146,3 +148,56 @@ def test_multiprocess_logging(verbose):
         assert "This is a warning message" in log_content
         assert "This is an error message" in log_content
         assert "This is a critical message" in log_content
+
+
+def test_get_read_sample_name():
+    with pytest.raises(ValueError, match="Invalid file extension.*"):
+        get_read_sample_name("qwdwqwdqwd")
+
+    with pytest.raises(ValueError, match="Invalid R1/R2 suffix."):
+        get_read_sample_name("qwdwqwdqwd.fq.gz")
+
+    assert get_read_sample_name("sample1_1.fq.gz") == "sample1"
+    assert get_read_sample_name("sample1_2.fq.gz") == "sample1"
+
+    assert get_read_sample_name("sample1_R1.fq.gz") == "sample1"
+    assert get_read_sample_name("sample1_R2.fq.gz") == "sample1"
+    assert get_read_sample_name("sample1_r1.fq.gz") == "sample1"
+    assert get_read_sample_name("sample1_r2.fq.gz") == "sample1"
+    assert get_read_sample_name("sample1.r1.fq.gz") == "sample1"
+    assert get_read_sample_name("sample1.r2.fq.gz") == "sample1"
+    assert get_read_sample_name("sample1.R1.fq.gz") == "sample1"
+    assert get_read_sample_name("sample1.R2.fq.gz") == "sample1"
+
+    assert get_read_sample_name("sample1_R1_L001.fq.gz") == "sample1_L001"
+    assert get_read_sample_name("sample1_R2_L001.fq.gz") == "sample1_L001"
```

Comment:
I am not sure if I like to keep the lane name here. And I know this is a  `get_read_sample_name` change, but I am unsure what value brings keeping lanes in sample names. Maybe is good somehow to keep it as a reminiscent of the sequencing design.

Reply:
My 2 cents is not to include the lane number in the sample name, since I suppose the same sample might be run across multiple lanes?

Reply:
That's a good point. However, when running across multiple lanes, we have this feature in the samplesheets that using the same ID across multiple rows with different libraries/indexes, you can pool reads together by concatenation.

Reply:
Aha. Cool. But then I think we really should just keep the actual sample name, stripping the lane part.

Reply:
Ok, it shouldn't be too difficult to implement, right @fbdtemme ?

Reply:
I would not remove anything except the R1/R2 identifier to create the default output sample names.
I think if we remove lane suffixes we create the impression that pixelator somehow "handles" this in a way.

I would prefer not to do anything smart with the name generation since that will always be quite brittle with regexes and needing to handle different naming conventions etc...

You can always use the new `--sample-name` option to override the default or just rename your input files to something sensible in the first place to get nicer output names by default.